### DISCUSSION TITLE
Agent Data Source Configuration fetch error

### DISCUSSION THREAD

**Discussion by Rashmitha677** [mindsphere/mindconnect-nodejs#313] [General]
Hello,

I received the below error:

Agent Data Source Configuration fetch error :  Error: Network error occurred request to https://southgate.eu1.mindsphere.io/api/agentmanagement/v3/oauth/token failed, reason: getaddrinfo EAI_AGAIN southgate.eu1.mindsphere.io:443 (hint: possible cause for this error is invalid date/time on the device) see also: https://opensource.mindsphere.io/docs/mindconnect-nodejs/troubleshooting.html
    at MindConnectAgent.<anonymous> (/home/sc2040/MindConnect/node_modules/@mindconnect/mindconnect-nodejs/dist/src/api/agent-auth.js:299:23)
    at throw (native)
    at rejected (/home/sc2040/MindConnect/node_modules/@mindconnect/mindconnect-nodejs/dist/src/api/agent-auth.js:6:65)
    at process._tickCallback (internal/process/next_tick.js:109:7)
    

 I had to generate new onboarding key to establish a connection to Mindsphere.  Could you please let us know what could have caused the issue?

**Comment by sn0wcat** [comment]
The `getaddrinfo EAI_AGAIN` error means that the device running the agent couldn't reach your DNS server to resolve the IP address of MindSphere. This can happen for multiple reasons, For example if you have problems with your network, if DNS is overloaded or offline etc. The library will actually try to recover from these problems, but if the network errors happen during the token renewal beyond the grace period of the old token and old [RAT](https://developer.mindsphere.io/howto/howto-agent-onboard.html#refreshing-a-registration) the exchange of the security information gets out of sync and the agent needs to be onboarded again.
